packages/web-pkg/src/composables/piniaStores/extensionRegistry/extensionRegistry.ts

```diff
@@ -11,40 +12,57 @@ export const useExtensionRegistry = defineStore('extensionRegistry', () => {
   const registerExtensions = (e: Ref<Extension[]>) => {
     extensions.value.push(e)
   }
-  const requestExtensions = <T extends Extension>(
-    type: ExtensionType,
-    options?: {
-      scopes?: ExtensionScope[]
-      extensionPointIds?: string[]
+  const requestExtensions = <T extends Extension>(options: {
+    extensionType?: ExtensionType
+    extensionPointIds?: string[]
+    extensionPoint?: ExtensionPoint
+  }) => {
```

Comment:
This feels inconsistent - why an array of ids but a single extension?

I don't think we query multiple `extensionPointId`s anywhere and conceptually it's not clear at all to me, why we would need it.. 

Moreover, I only see only one occurrence of requesting only by type (in search) - do we really need that? Wouldn't it be more consistent to use an extension point there too? If we need to request extensions by type at all - I'd argue that would justify an independent function `requestExtensionsByType` - to me it feels this function does too much as it is implemented right now.

My suggestion would be (basically as I wrote in my original comment):
```typescript
// add a type parameter to ExtensionPoint interface
export type ExtensionPoint<T> = {
…
}

// specify it when defining the extension point
export const batchActionsExtensionPoint: ExtensionPoint<ActionExtension> = {
  id: 'app.files.batch-actions',
  extensionType: 'action',
  multiple: true
}

// infer extension type from ExtensionPoint param
const requestExtensions = <T>(extensionPoint: ExtensionPoint<T>): T[] => {
…
}

// batchActions is of type ActionExtension[]
const batchActions = requestExtensions(batchActionsExtensionPoint)
```

P.S.: this would make having the extension points in web-pkg even more handy

Reply:
I don't get it. As of now it's still valid to request extensions by extension type, without any extensionPoint. E.g. we currently don't have a `search` extension point, we just request extensions of type `search`. Also, if I develop an extension and two apps that want to use the extension, where do I define the `extensionPoint`? Either results in lots of code duplication or the necessity to have a shared package. Being able to request extensions simply by an extensionType or an extensionPointId seems to be a lot more flexible.

Reply:
Could you please check again? Looks a lot cleaner now, thanks for the input ❤️